pkgs/applications/virtualization/conmon/default.nix

```
@@ -0,0 +1,39 @@
+{ flavor ? ""
+, ldflags ? ""
+, stdenv
+, fetchFromGitHub
+, git
+, pkgconfig
+, glib
+, glibc
+, systemd
+}:
+
+stdenv.mkDerivation rec {
+  project = "conmon";
+  name = "${project}-${version}${flavor}";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = project;
+    rev = "v${version}";
+    sha256 = "08fgkbv7hq62lcq39za9pm2s2j92ismgrkvfm7acwbvajqh9syjb";
+  };
+
+  nativeBuildInputs = [ git pkgconfig ];
+  buildInputs = [ glib systemd ] ++
+                stdenv.lib.optionals (glibc != null) [ glibc glibc.static ];
+
+  makeFlags = [ "LDFLAGS=${ldflags}" ];
+
+  installPhase = "install -Dm755 bin/${project} $out/bin/${project}${flavor}";
+
+  meta = with import <nixpkgs/lib>; {
```

Use `with stdenv.lib` here instead. Usage of `<nixpkgs>` within the upstream `pkgs` subsystem is (or should be!) more or less banned outside of extremely specific cases, because it means the derivation now has an impure input, namely whatever your `NIX_PATH` or default `nixpkgs` channel is set to at evaluation time. In other words, this is using a potentially completely different `nixpkgs` version than the one that is used to define the package itself.

```suggestion
  meta = with stdenv.lib; {
```

---

Yes I totally agree. Changed it as adapted.